code/modules/awaymissions/corpse.dm

```diff
@@ -4,294 +4,474 @@
 
-/obj/effect/landmark/corpse/bridgeofficer
+/obj/effect/mob_spawn/human/bridgeofficer
 	name = "Bridge Officer"
-	corpseradio = /obj/item/device/radio/headset/heads/hop
-	corpseuniform = /obj/item/clothing/under/rank/centcom_officer
-	corpsesuit = /obj/item/clothing/suit/armor/bulletproof
-	corpseshoes = /obj/item/clothing/shoes/black
-	corpseglasses = /obj/item/clothing/glasses/sunglasses
-	corpseid = 1
-	corpseidjob = "Bridge Officer"
-	corpseidaccess = "Captain"
-
-/obj/effect/landmark/corpse/commander
+	id_job = "Bridge Officer"
+	id_access = "Captain"
+	outfit = /datum/outfit/nanotrasenbridgeofficercorpse
+
+/datum/outfit/nanotrasenbridgeofficercorpse
+	name = "Bridge Officer Corpse"
+	l_ear = /obj/item/device/radio/headset/heads/hop
+	uniform = /obj/item/clothing/under/rank/centcom_officer
+	suit = /obj/item/clothing/suit/armor/bulletproof
+	shoes = /obj/item/clothing/shoes/black
+	glasses = /obj/item/clothing/glasses/sunglasses
+	id = /obj/item/weapon/card/id
+
+
+/obj/effect/mob_spawn/human/commander
 	name = "Commander"
-	corpseuniform = /obj/item/clothing/under/rank/centcom_commander
-	corpsesuit = /obj/item/clothing/suit/armor/bulletproof
-	corpseradio = /obj/item/device/radio/headset/heads/captain
-	corpseglasses = /obj/item/clothing/glasses/eyepatch
-	corpsemask = /obj/item/clothing/mask/cigarette/cigar/cohiba
-	corpsehelmet = /obj/item/clothing/head/centhat
-	corpsegloves = /obj/item/clothing/gloves/combat
-	corpseshoes = /obj/item/clothing/shoes/combat
-	corpsepocket1 = /obj/item/weapon/lighter/zippo
-	corpseid = 1
-	corpseidjob = "Commander"
-	corpseidaccess = "Captain"
-
-/obj/effect/landmark/corpse/abductor //Connected to ruins, for some reason?
+	id_job = "Commander"
+	id_access = "Captain"
+	outfit = /datum/outfit/nanotrasencommandercorpse
+
+/datum/outfit/nanotrasencommandercorpse
+	name = "Commander Corpse"
+
+	uniform = /obj/item/clothing/under/rank/centcom/officer
+	gloves =  /obj/item/clothing/gloves/color/white
+	shoes = /obj/item/clothing/shoes/centcom
+	head = /obj/item/clothing/head/beret/centcom/officer
+	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
+	id = /obj/item/weapon/card/id/centcom
+
+
+
+/obj/effect/mob_spawn/human/abductor
 	name = "abductor"
-	mobname = "???"
-	mob_species = "abductor"
-	corpseuniform = /obj/item/clothing/under/color/grey
-	corpseshoes = /obj/item/clothing/shoes/combat
+	mob_name = "alien"
+	mob_species = /datum/species/abductor
+	outfit = /datum/outfit/abductorcorpse
+
+/datum/outfit/abductorcorpse
+	name = "Abductor Corpse"
+	uniform = /obj/item/clothing/under/color/grey
+	shoes = /obj/item/clothing/shoes/combat
+
+//For ghost bar.
+/obj/effect/mob_spawn/human/alive/space_bar_patron
+	name = "Bar cryogenics"
+	mob_name = "Bar patron"
+	random = TRUE
+	permanent = TRUE
+	uses = -1
+	outfit = /datum/outfit/spacebartender
+	assignedrole = "Space Bar Patron"
+
+/obj/effect/mob_spawn/human/alive/space_bar_patron/attack_hand(mob/user)

```

Comment:
Why is this defined on this very specific subtype of the mob_spawn, rather than the mob_spawn parent?

Reply:
/obj/effect/mob_spawn/human/alive refers to the spawner object. in the case a cryogenic pod.  /obj/effect/mob_spawn/human/alive/space_bar_patron defines what its spawning. its spawning a living thing rather a corpse here.